api/v1beta1/addresspool_webhook.go

```diff
@@ -138,3 +143,14 @@ func getAddressPoolCIDRs(addressPool *AddressPool) ([]*net.IPNet, error) {
 	}
 	return CIDRs, nil
 }
+
+func validateBGPAdvertisements(bgpAdvertisements []BgpAdvertisement) error {
+	for i := 0; i < len(bgpAdvertisements); i++ {
+		for j := i + 1; j < len(bgpAdvertisements); j++ {
+			if reflect.DeepEqual(bgpAdvertisements[i], bgpAdvertisements[j]) {
+				return errors.New("duplicate definition of bgpadvertisement")
+			}
+		}
+	}
```

Comment:
don't u need to compare against existing adv as well ?

Reply:
if you have an adv in an already configured addresspool and you add it again it's not a duplicate as you add it just once in the yaml.

Reply:
and that could happen as we update the yaml and lead to false +ve, Cool!!